File: solved.ac/class2/15829/15829.py
import sys

input = sys.stdin.readline
print = lambda x: sys.stdout.write(str(x) + '\n')

len = int(input())
s = input()[0:len]

r = 31
m = 1234567891

sum = 0

for index in range(len):
  id = ord(s[index]) % 96
  coef = pow(r, index)
  sum += id * coef % m

result = sum % m

print(result)

File: solved.ac/class2/15829/15829.rb
len = gets.to_i
str = gets.slice(0, len)

r = 31
m = 1234567891

sum = 0

for index in 0..len-1 do
  id = str[index].ord % 96
  coef = r ** index
  sum = sum + (id * coef % m)
end

result = sum % m

puts result